tests/components/compit/test_config_flow.py

```diff
@@ -0,0 +1,191 @@
+    ):
+        result = await flow.async_step_user(
+            {CONF_EMAIL: "test@example.com", CONF_PASSWORD: "password"}
+        )
+
+    assert result["type"] == FlowResultType.FORM
+    assert result["errors"] == {"base": "invalid_auth"}
+
+
+async def test_async_step_user_cannot_connect(hass: HomeAssistant) -> None:
+    """Test user step with connection error."""
+    flow = CompitConfigFlow()
+    flow.hass = hass
+
+    with patch(
+        "homeassistant.components.compit.config_flow.CompitAPI.authenticate",
+        side_effect=CannotConnect,
+    ):
+        result = await flow.async_step_user(
+            {CONF_EMAIL: "test@example.com", CONF_PASSWORD: "password"}
+        )
+
+    assert result["type"] == FlowResultType.FORM
+    assert result["errors"] == {"base": "cannot_connect"}
+
+
+async def test_async_step_user_unknown_error(hass: HomeAssistant) -> None:
+    """Test user step with unknown error."""
+    flow = CompitConfigFlow()
+    flow.hass = hass
+
+    with patch(
+        "homeassistant.components.compit.config_flow.CompitAPI.authenticate",
+        side_effect=Exception,
+    ):
+        result = await flow.async_step_user(
+            {CONF_EMAIL: "test@example.com", CONF_PASSWORD: "password"}
+        )
+
+    assert result["type"] == FlowResultType.FORM
+    assert result["errors"] == {"base": "unknown"}
+
+
+@pytest.fixture
+def mock_reauth_entry():
+    """Return a mock config entry."""
+    return config_entries.ConfigEntry(
+        version=1,
+        domain=DOMAIN,
+        title="Compit",
+        data={CONF_EMAIL: "test@example.com"},
+        source=config_entries.SOURCE_REAUTH,
+        entry_id="1",
+        unique_id="compit_test@example.com",
+        discovery_keys={},
+        minor_version=0,
+        options={},
+    )
+
+
+async def test_async_step_reauth_confirm_success(
+    hass: HomeAssistant, mock_reauth_entry: config_entries.ConfigEntry
+) -> None:
+    """Test reauth confirm step with successful authentication."""
+    hass.config_entries._entries[mock_reauth_entry.entry_id] = mock_reauth_entry
+
+    flow = CompitConfigFlow()
+    flow.hass = hass
+    flow._get_reauth_entry = Mock(return_value=mock_reauth_entry)
+    flow.context = {"source": config_entries.SOURCE_REAUTH}
+
+    with (
+        patch(
+            "homeassistant.components.compit.config_flow.CompitAPI.authenticate",
+            return_value=SystemInfo(gates=[]),
+        ),
+        patch.object(flow, "async_set_unique_id", return_value=None),
+        patch.object(flow, "_abort_if_unique_id_mismatch", return_value=None),
+        patch.object(flow, "_abort_if_unique_id_configured", return_value=None),
+    ):
+        result = await flow.async_step_reauth_confirm(
+            {CONF_PASSWORD: "new_password", CONF_EMAIL: "test@example.com"}
+        )
+
+    assert result["type"] == FlowResultType.ABORT
+    assert result["reason"] == "reauth_successful"
+
+
+async def test_async_step_reauth_confirm_invalid_auth(
```

Comment:
test_async_step_reauth_confirm_cannot_connect and test_async_step_reauth_confirm_unknown_error can be merged into this one with test params